lib/std/os.zig

```diff
@@ -3190,3 +3190,33 @@ pub fn sched_yield() SchedYieldError!void {
         else => return error.SystemCannotYield,
     }
 }
+
+// TODO support for non-null terminated strings?
+pub fn memfdCreateC(name: [*]const u8, flags: usize) !fd_t {
+    if (builtin.os != .linux) @compileError("memfd_create() not implemented for this target");
+
+    const rc = linux.memfd_create(name, flags);
+    switch (linux.getErrno(rc)) {
+        0 => return @intCast(fd_t, rc),
+        EFAULT => unreachable, // name has invalid memory
+        EINVAL => unreachable, // name/flags are faulty
+        ENFILE => return error.SystemFdQuotaExceeded,
+        EMFILE => return error.ProcessFdQuotaExceeded,
+        ENOMEM => return error.OutOfMemory,
+        else => |err| return unexpectedErrno(err),
+    }
+}
+
+fn toMemFdPath(name: []const u8) ![250]u8 {
```

Comment:
Could you factor out the magic 250 constant into `MFD_NAME_MAX_LEN` (to match the linux kernel?)